1config-ui/src/com/brunobonacci/oneconfig/ui/controller.cljs

```diff
@@ -78,6 +81,56 @@
   (swap! state assoc :1config-version version))
 
 
+(defn GETAsync [url]
+  (let [out (chan)
+        handler #(put! out %1)]
+    (GET url {:handler handler
+              :format          :json
+              :response-format :json
+              :keywords?       true
+              :error-handler handler})
+    out))
+
+(defn create-url-a [item]
+  (let [{:keys [key env version change-num]} item]
+    (gs/format "/configs/keys/%s/envs/%s/versions/%s?change-num=%s"
+               (gs/urlEncode key) (gs/urlEncode env)
+               (gs/urlEncode version) change-num)))
+
+(defn compare-selected-items [compare-items]
+  (go
+    (let [urls (map #(create-url-a %) (get compare-items :items-meta))
+          configs [(<! (GETAsync (first urls)))
+                   (<! (GETAsync (last urls)))]
+          selected-entries (map #(:encoded-value %) configs)]
+      (swap! state #(-> %
+                        (assoc-in [:new-entry] empty-new-entry)
+                        (assoc-in [:selected :entries]  selected-entries)
+                        (assoc-in [:entry-management-button-style] "ui inverted disabled button")
+                        (assoc-in [:client-mode] :compare-entry-mode))))))
```

Comment:
Not sure why we need `core.async` here as you could do exactly the same without.
instead of `GETAsync` we could reuse `get-config-item!` which is already async (see the response handlers)
and just add the entries in a specific place like your `[:selected :entries]`

the UI will only render the diff page if both entries values have been set.

Reply:
since AceDiff has the concept of `left` and `right` parts for the diff then we could update the state to deliver the entries in: `[:selected :entries :left]` and `[:selected :entries :right]`

Reply:
Ok, I'll definitely try it, thanks!

Reply:
for example it is possible to change `get-config-item!` like the following:

```
(defn get-config-item!
  [item & {:keys [result-path client-mode]
           :or {result-path [:item-data] client-mode :show-entry-mode}}]
  (let [{:keys [key env version change-num content-type]} item
        get-url (gs/format "/configs/keys/%s/envs/%s/versions/%s?change-num=%s"
                           (gs/urlEncode key) (gs/urlEncode env)
                           (gs/urlEncode version) change-num)]

    ;; FIXME: not sure why we need this
    (swap! state assoc-in [:item-params] {:key          key
                                       :env          env
                                       :version      version
                                       :change-num   change-num
                                       :content-type content-type})

     ;; but this is the key, when the response arrive we tell where to put it in the state
     ;; by specifying the `result-path`
    (GET get-url {:handler         (fn [response]
                                     (swap! state
                                            (fn [s]
                                              (-> s
                                                   ;; maybe here rather than having just the value
                                                   ;; we should keep the whole config entry
                                                  (assoc-in result-path (get response :encoded-value))
                                                  (assoc :client-mode client-mode)))))
                  :format          :json
                  :response-format :json
                  :keywords?       true
                  :error-handler   backend-error-handler})))
```

This implementation is functionally equivalent to the current implementation and also backwards compatible.
However, it offers the possibility to execute requests such as:

```
(get-config-item! entry1 :result-path [:selected :entries :left])
(get-config-item! entry2 :result-path [:selected :entries :right])
```
The ui, then will render the diff only when both: `:left` and `:right` are non `nil`.